File: ewbapp/Models/Enums/InfestationSize.swift
import Foundation

enum InfestationSize: String, CaseIterable, Codable {
    case small = "small"
    case medium = "medium"
    case large = "large"

    var displayName: String {
        switch self {
        case .small: return "Small"
        case .medium: return "Medium"
        case .large: return "Large"
        }
    }

    var areaDescription: String {
        switch self {
        case .small: return "< 5 m²"
        case .medium: return "5 – 50 m²"
        case .large: return "> 50 m²"
        }
    }
}

File: app/src/main/java/org/yac/llamarangers/domain/enums/InfestationSize.kt
package org.yac.llamarangers.domain.enums

enum class InfestationSize(val displayName: String, val areaDescription: String) {
    SMALL("Small", "< 5 m²"),
    MEDIUM("Medium", "5–50 m²"),
    LARGE("Large", "> 50 m²");

    companion object {
        fun fromRaw(raw: String): InfestationSize =
            entries.firstOrNull { it.name == raw } ?: SMALL
    }
}
